### PR TITLE
Optional non-atomic move for use with Docker/cross-device moves

### PR DESCRIPTION
**Pull request by haggholm**
See #9863

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,6 +61,7 @@
     "lodash.uniq": "~3.2.2",
     "lodash.without": "~3.2.1",
     "mkdirp": "~0.5.1",
+    "node-fs-extra": "~0.8.1",
```
Comment (vsimonian) [inline package.json]: I think you meant to require `fs-extra` here and in the code as well - `node-fs-extra` points to an old fork for which the linked source repository doesn't exist anymore.

**Comment by othiym23** [comment]
As described on #9683, this is not a good solution to the problem, because it doesn't address the underlying race condition the `fs.rename()` is intended to solve, where multiple installs could end up writing to the install target at the same time, leaving the installed package tree in an inconsistent or corrupt state. We need a solution for Docker users, but at this point I think that has more to do with how and where Docker mounts the filesystem used by npm, and not in npm itself.

That aside, thanks for taking the time to put this together!

**Comment by graingert** [comment]
@othiym23 do you mean #9863

**Comment by othiym23** [comment]
Yes! Thanks for the heads up – corrected inline.

**Comment by frank-dspeed** [comment]
DOCKERFILE THAT APPLYS RENAME TO BE MOVE !!!! FOR DOCKER USERS use it if you understand this ISSUS:

~~~
FROM node:5.6
WORKDIR /usr/local/lib/node_modules/npm
RUN npm install --save fs-extra
RUN sed -i s/graceful-fs/fs-extra/g /usr/local/lib/node_modules/npm/lib/utils/rename.js
RUN sed -i s/fs.rename/fs.move/g /usr/local/lib/node_modules/npm/lib/utils/rename.js
WORKDIR /app
#...... main part of the Dockerfile greetings.
~~~

**Pull request closed without merging**